src/kudu/consensus/log_index.cc

```diff
@@ -193,23 +230,115 @@ Status LogIndex::OpenAllChunksOnStartup(Env *env) {
     VLOG(1) << "Opening index file on startup: " << fname << " for chunk idx " << chunk_idx;
 
     scoped_refptr<IndexChunk> chunk;
-    RETURN_NOT_OK(OpenAndInsertChunk(chunk_idx, &chunk));
+    RETURN_NOT_OK(OpenAndInsertChunk(chunk_idx, &chunk, /*should_mmap=*/false));
+  }
+
+  // mmap 'kNumChunksToMmap' chunks. Note that the latest chunks are mmaped
+  // (chunks having the highest chunk_idx)
+  int64_t mmapped_chunks = 0;
+  for (auto rit = open_chunks_.rbegin(); rit != open_chunks_.rend(); ++rit) {
+    if (mmapped_chunks == kNumChunksToMmap) {
+      break;
+    }
+
+    RETURN_NOT_OK(rit->second->Mmap());
+    mmapped_chunks++;
+  }
+
+  return Status::OK();
+}
+
+void LogIndex::SetNumMmapChunks(int64_t num_chunks) {
+  if (num_chunks <= 0)
+    return;
+
+  std::lock_guard<simple_spinlock> l(open_chunks_lock_);
+  kNumChunksToMmap = num_chunks;
+
+  // If necessary, unmap additional chunks
+  if (open_chunks_.size() <= kNumChunksToMmap) {
+    return;
+  }
+
+  // get the number of chunks that are currently mmapped
+  int64_t num_chunks_mmapped = 0;
```

Comment:
can we abstract out this into a function, as we might have to reset book-keeping of mmapped_chunks.

Reply:
Thought about this, but the current places where it used has slightly different semantic (read all entries v/s read until the first mmapped entry) and abstracting does not add much value. I think I will retain it for now?